fastlane/fastlane
# [trainer] fix issues where number of failures would always be zero

### Checklist

### Motivation and Context

Closes #21565

### Description
If the test case name has spaces or any special characters, the reported `number_of_failures` will always be 0. This will also not include any failed tests inside the junit report.

The reason this was happening was because during the comparison to check the test failure there was a sanitization step that's not applied to the identifier itself so it never matched.

This is from the log output of such a test.
```
** TEST FAILED **
[15:11:03]: Exit status: 65
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 3 |
| Number of failures | 0 |
+--------------------+---+

[15:11:04]: Test execution failed. Exit status: 65
```

### Testing Steps

The easiest way to reproduce it was to use https://github.com/Quick/quick because Quick 7 produces test case names with spaces if the spec names had spaces in it. And when you run `scan` the output will always say the number of failures is 0.

I have attached a sample project.

1. Download project
2. Run `bundle install`
3. Run `bundle exec fastlane test` 

When it runs the output will show number of failures as 0.

Then you can uncomment the branch line in the `Gemfile` and run bundle install and switch to this branch. Then you will see that the number of failure is 1.

[SpaceTests.zip](https://github.com/fastlane/fastlane/files/12221320/SpaceTests.zip)
